File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EInverseSurfaceDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInverseSurfaceDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd7
            AppTheme.STANDARD -> R.color.cstd7
            AppTheme.BLUE -> R.color.cbd7
            AppTheme.GREEN -> R.color.cgd7
            AppTheme.RED -> R.color.crd7
            AppTheme.NOTHING -> R.color.cnod7
        }
    }
}

File: lib/layer2_core/dark/e_inverse_surface_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EInverseSurfaceDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd7;
      case 'standard': return cstd7;
      case 'blue': return cbd7;
      case 'green': return cgd7;
      case 'red': return crd7;
      case 'nothing': return cnod7;
      default: return cstd7;
    }
  }
}
